react/src/components/elements/payment-method/PaymentMethod.tsx

```diff
@@ -42,27 +44,32 @@ export const PaymentMethod = forwardRef<
 >(({ children, className, portal, ...rest }, ref) => {
   const props = resolveDesignProps(rest);
 
-  const { settings, layout, setLayout } = useEmbed();
+  const { data, settings, stripe, layout, setLayout } = useEmbed();
 
   const paymentMethod = useMemo(() => {
-    /**
-     * @TODO: Replace this with payment details from data context
-     */
-    const cardNumber = "4512";
-    const expirationDate = "2024-12-22";
-
-    const timeToExpiration = Math.round(
-      +new Date(expirationDate) - +new Date(),
-    );
-    const monthsToExpiration = Math.round(
-      timeToExpiration / (1000 * 60 * 60 * 24 * 30),
-    );
+    const { cardLast4, cardExpMonth, cardExpYear } =
+      data.subscription?.paymentMethod || {};
+
+    let monthsToExpiration: number | undefined;
+    if (typeof cardExpYear === "number" && typeof cardExpMonth === "number") {
+      const timeToExpiration = Math.round(
+        // TODO: is `cardExpMonth` a zero-based index?
```

Comment:
I think it is 1-based, so 1=January, 12=December.

It's this value: https://docs.stripe.com/api/payment_methods/object#payment_method_object-card-exp_month